### DISCUSSION TITLE
Introducing paymaster-examples: Your Gateway to Gas Fee Flexibility!

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#51] [Announcements]
![](https://github.com/matter-labs/paymaster-examples/blob/main/paymaster-examples.png?raw=true)
We're thrilled to announce [paymaster-examples](https://github.com/matter-labs/paymaster-examples), **an open-source repository packed with all the information you need to dive into the world of paymasters.**

Paymasters offer application developers the power to revolutionise gas fee handling for their users. Whether it's enabling gasless transactions or facilitating fee payment in specific tokens, paymasters bring unparalleled flexibility. **Thanks to the [native account abstraction implementation on zkSync Era](https://era.zksync.io/docs/reference/concepts/account-abstraction.html), paymasters can now be utilised by both EOAs and smart contract accounts.**

Inside the repository, you'll find four distinct paymasters:

- 🆓 **Gasless Paymaster**: Seamlessly covers fees for any account.
- 📜 **Allowlist Paymaster**: Handles fees exclusively for accounts listed in a predefined "allow list."
- 🎫 **ERC20 Fixed Paymaster**: Accepts a fixed amount of a specified ERC20 token as payment for gas fees. It caters exclusively to accounts holding the designated token balance.
- 🎨 **ERC721 Gated Paymaster**: Handles fees for accounts that possess a particular ERC721 token (NFT).

Additionally, we've included a frontend that serves as a reference for integrating paymasters into your applications and allows you to test them effectively.

We have plans to expand the repository with more examples but we also invite all developers [to contribute to this open source initiative]((https://github.com/matter-labs/paymaster-examples/blob/main/CONTRIBUTING.md)).

Visit the [Paymaster examples repo](https://github.com/matter-labs/paymaster-examples) to explore the possibilities of gas fee management.

**Comment by VietTran18** [comment]
good

**Comment by jommi9** [comment]
Very cool! Cant wait to only hold LSTs on ZKSync and use them as gas!

**Comment by digijoe7** [comment]
Love it!

**Comment by aligh98** [comment]
Looks great. waiting for more details

**Comment by kimberleeni** [comment]
This announcement introduces paymaster-examples, a repository containing various paymasters for flexible gas fee management on zkSync Era. Developers can explore gasless transactions, allowlist-based fee handling, ERC20 fixed fee payments, and ERC721 token-gated fee management. The repository also includes a frontend for testing and integration reference. This initiative empowers developers to enhance user experience and drive innovation in gas fee management.

**Comment by neumagbutorcapri** [comment]
Explore gas fee management possibilities with paymaster-examples! Revolutionize gas handling for users with gasless transactions, token-specific fee payment, and more. Dive in and contribute to this open-source initiative today!  #PaymasterExamples #GasFeeManagement #OpenSource

**Comment by singletonbrittany** [comment]
Exciting news! The paymaster-examples repository is now live, offering developers a comprehensive guide to leveraging paymasters for enhanced gas fee management. Dive in, explore the examples, and revolutionize your application's fee handling capabilities! #OpenSource #GaslessTransactions #zkSyncEra

**Comment by sanddethe** [comment]
The paymaster-examples repository provides developers with a range of paymaster options for gas fee management on zkSync Era. These paymasters offer flexibility and can be utilised by both EOAs and smart contract accounts. The repository includes a frontend for testing and serves as a reference for integrating paymasters into applications. Developers are also invited to contribute to this open-source initiative.
